client/src/livingdocs-component-app/app.js

```diff
@@ -0,0 +1,314 @@
+import { inject, singleton } from 'aurelia-framework';
+import { I18N } from 'aurelia-i18n';
+import ItemStore from 'resources/ItemStore.js';
+import QTargets from 'resources/QTargets.js';
+import qEnv from 'resources/qEnv.js';
+
+@singleton()
+@inject(ItemStore, QTargets, I18N)
+export class App {
+
+  constructor(itemStore, qTargets, i18n) {
+    this.itemStore = itemStore;
+    this.qTargets = qTargets;
+    this.i18n = i18n;
+  }
+
+  activate() {
+    this.selectedItems = [];
+
+    const paramsQuery = /params=(.*)&?/.exec(window.location.search);
+    const targetQuery = /target=([^&]*)/.exec(window.location.search);
+
+    try {
+      this.selectedItem = JSON.parse(decodeURIComponent(paramsQuery[1]));
+      this.selectedItems.push(this.selectedItem);
+    } catch (e) {
+      // nevermind an error here, if there is no valid config, we handle it like there is none at all
+    }
+
+    try {
+      this.targetKey = decodeURIComponent(targetQuery[1]);
+    } catch (e) {
+      // nevermind for the moment - return error message of missing target later on
+    }
+  }
+
+  async attached() {
+    await this.loadTarget();
+    await this.reloadItems();
+    this.loadView();
+  }
+
+  async loadTarget() {
+    try {
+      const targets = await this.qTargets.get('availableTargets');
+      this.target = targets.filter(target => {
+        return target.key === this.targetKey;
+      })[0];
+    } catch (e) {
+      // nevermind, we'll check if target exists before loading preview
+    }
+  }
+
+  loadView() {
+    if (this.selectedItem) {
+      this.title = this.items
+        .filter(item => {
+          return item.id === this.selectedItem.id;
+        })
+        .map(item => {
+          return item.conf.title;
+        })[0];
+
+      if (this.target && this.selectedItem) {
+        this.loadPreview();
+      }
+      this.loadDisplayOptions();
+    }
+  }
+
+  selectItem(item) {
+    this.loadSelectedItem(item);
+    this.loadView();
+  }
+
+  loadSelectedItem(item) {
+    this.title = item.conf.title;
+    const alreadySelected = this.selectedItems.filter(selectedItem => {
+      return item.conf._id === selectedItem.id;
+    });
+
+    if (alreadySelected.length > 0) {
+      this.selectedItem = alreadySelected[0];
+    } else {
+      this.selectedItem = {
+        id: item.conf._id,
+        toolRuntimeConfig: {
+          displayOptions: {}
+        }
+      };
+    }
+  }
+
+  changeDisplayOptions() {
```

Comment:
I am a little confused by this function and it's name. As far as I get it, it pushes the `selectedItem` to the end of the `selectedItems` array and then triggers a preview loading. But what does this have to do with changing display options? Can you explain?